app/controllers/concerns/uploadable.rb

```diff
@@ -14,7 +14,7 @@ def process_upload_background(args = {})
 
     job_logger.write 'Enqueueing job to start in the background.'
 
-    uid = is_api? ? job_logger.id : upload_params[:item_id].to_i
+    uid = is_api? ? job_logger.id : upload_params[:item_id]
```

Comment:
the whole reason of being for this parameter is to track the job. Why is .id used? Shouldn't it be .uid?